src/System.Text.Primitives/System/Text/Encoding/Utf8/Utf8EncodedCodePoint.cs

```diff
@@ -33,6 +33,27 @@ public Utf8EncodedCodePoint(char character) : this()
             }
         }
 
+        public Utf8EncodedCodePoint(UnicodeCodePoint codePoint) : this()
+        {
+            if (UnicodeCodePoint.IsSurrogate(codePoint))
+            {
+                throw new ArgumentOutOfRangeException("character", "Surrogate characters are not allowed");
+            }
+
+            unsafe
+            {
+                fixed (byte* encodedData = &_byte0)
+                {
+                    var buffer = new Span<byte>(encodedData, 4);
+                    if (!Utf8Encoder.TryEncodeCodePoint(codePoint, buffer, out _length))
```

Comment:
Is this class going away as the TODO at the top says? It does not really matter what it does if we are going to delete it anyway...